networking/eth2/src/main/java/tech/pegasys/artemis/networking/eth2/gossip/AttestationTopicSubscriptions.java

```diff
@@ -0,0 +1,55 @@
+/*
+ * Copyright 2020 ConsenSys AG.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+
+package tech.pegasys.artemis.networking.eth2.gossip;
+
+import static com.google.common.primitives.UnsignedLong.ZERO;
+import static tech.pegasys.artemis.datastructures.util.BeaconStateUtil.max;
+
+import com.google.common.primitives.UnsignedLong;
+import java.util.HashMap;
+import java.util.Iterator;
+import java.util.Map;
+import java.util.Map.Entry;
+import tech.pegasys.artemis.networking.eth2.Eth2Network;
+import tech.pegasys.artemis.util.time.channels.SlotEventsChannel;
+
+public class AttestationTopicSubscriptions implements SlotEventsChannel {
+  private final Map<Integer, UnsignedLong> unsubscriptionSlotByCommittee = new HashMap<>();
+  private final Eth2Network eth2Network;
+
+  public AttestationTopicSubscriptions(final Eth2Network eth2Network) {
+    this.eth2Network = eth2Network;
+  }
+
+  public synchronized void subscribeToCommittee(
+      final int committeeIndex, final UnsignedLong aggregationSlot) {
+    eth2Network.subscribeToAttestationCommitteeTopic(committeeIndex);
+    final UnsignedLong currentUnsubscribeSlot =
+        unsubscriptionSlotByCommittee.getOrDefault(committeeIndex, ZERO);
```

Comment:
Is this due to potentially subscribing to the same topic for a later slot?

Reply:
Nvm, yeah it seems pretty clear.

Reply:
Yes. I think it's most likely to happen if we happen when one of the validators we run is aggregator for a committee in this epoch and another one (or the same one) is aggregator for that same committee next epoch.  You'd want to stay subscribed until the second one is reached.